aten/src/ATen/native/LegacyDefinitions.cpp

```diff
@@ -51,15 +51,33 @@ Tensor& zero_(Tensor& self) {
 }
 
 Tensor & masked_fill_(Tensor& self, const Tensor & mask, Scalar value) {
-  return at::legacy::th::_th_masked_fill_(self, mask, value);
+  if (mask.dtype() == at::ScalarType::Byte) {
+    AT_WARN("masked_fill_ received a mask with dtype torch.uint8, this behavior is now deprecated," \
```

Comment:
I would avoid putting in the warnings for now until we change the default of comparison ops -- otherwise you force users to write unnecessary `.to(torch.bool)` in a number of places.